enterprise/frontend/src/metabase-enterprise/data-studio/data-model/components/TableSection/TableSection.tsx

```diff
@@ -278,23 +278,34 @@ const TableSectionBase = ({
 
           {!hasFields && <EmptyState message={t`This table has no fields`} />}
 
+          {/* NOTE: We're using here CSS display property to avoid scroll jump when toggling sorting mode. */}
           {hasFields && (
             <>
-              {isSorting && (
+              <Box
+                style={{
+                  display: isSorting ? "block" : "none",
+                }}
+                aria-hidden={!isSorting}
+              >
                 <TableSortableFieldList
                   activeFieldId={activeFieldId}
                   table={table}
                   onChange={handleCustomFieldOrderChange}
                 />
-              )}
+              </Box>
 
-              {!isSorting && (
+              <Box
+                style={{
+                  display: !isSorting ? "block" : "none",
```

Comment:
[nit of nits](https://refactoring.com/catalog/reverseConditional.html): 
```suggestion
                  display: isSorting ? "none" : "block",
```